jax/experimental/compilation_cache/file_system_cache.py

```diff
@@ -37,6 +39,42 @@ def put(self, key: str, value: bytes):
     """Adds new cache entry, possibly evicting older entries."""
     if not key:
       raise ValueError("key cannot be empty")
-      #TODO(colemanliyah):implement eviction
-    with open(os.path.join(self._path, key), "wb") as file:
+
+    directory_size = self.get_size_of_directory()
+    new_file_size = 0
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+      new_file_size = self.creating_a_file(tmpdir, key, value)
```

Comment:
This is a clever way to get the size of the file before writing it to the cache directory! I think it's a little overkill here though, and we can just assume the new file size is equal to `len(value)` (i.e. the byte size of `value`). Writing to the temporary directory will also take extra time and potentially use too much disk space if `value` is large too.